[Store Alerts] Adjust inner styles

psealock opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/47034

This PR updates the Store Alert styles.

Before
<img width="739" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/e07383d8-8b15-433d-8b94-d4bf82ddbd6c">


After
<img width="717" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/df7f6657-d2f8-49f2-9cae-2ed90b964a5e">


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Make sure you have more than one Store Alert visible

1. Trigger multiple notes. First see what's already in your DB.
```sql
SELECT note_id,is_read,is_deleted,status,type from wp_wc_admin_notes WHERE type = 'update'
```
2. Make a few "unactioned".
```sql
UPDATE wp_wc_admin_notes SET status = 'unactioned' WHERE note_id IN (1,2,3,4);
```
3. You may need to mark some as not deleted.
```sql 
UPDATE wp_wc_admin_notes SET is_deleted = 0 WHERE note_id IN (1,2,3,4);
```
4. See alerts on various screens like Home, Customers, Settings and be sure to check various sizes.
5. Compare to designs found at oDTJrNy6IXwdgFNinZmYgI-fi

<img width="822" alt="Main component" src="https://github.com/woocommerce/woocommerce/assets/5121465/cc0b9881-7469-471f-add8-ef53d8f72c52">

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Update Store Alert styles

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

psealock committed:
apply CSS

psealock committed:
adjust margin

psealock committed:
cleanup

psealock committed:
adjust gap between buttons

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

psealock committed:
Merge branch 'update/store-alerts-inner-styles' of github.com:woocommerce/woocommerce into update/store-alerts-inner-styles

chihsuan commented on plugins/woocommerce-admin/client/layout/store-alerts/index.js:
```diff
@@ -84,6 +84,8 @@ export const StoreAlerts = () => {
 	} );
 
+	console.log( alerts );
+
```
```suggestion
```

chihsuan commented on plugins/woocommerce-admin/client/layout/store-alerts/style.scss:
```diff
@@ -82,30 +84,34 @@
 
 	@include breakpoint("<782px") {
 		margin-bottom: $gap-large;
 		padding: $gap;
```
This overrides the default padding for small screens. I'm unsure if we need this because the design doesn't have small screens. 🤔  Just raising it here for heads up. 

<img width="593" alt="Screenshot 2024-07-05 at 13 58 11" src="https://github.com/woocommerce/woocommerce/assets/4344253/ab866d57-2875-4c00-9d98-0f9f99e0313d">

chihsuan commented on plugins/woocommerce-admin/client/layout/store-alerts/style.scss:
```diff
@@ -82,30 +84,34 @@
 	}
 }
 
+.woocommerce-store-alerts__header {
```
```suggestion
```

chihsuan reviewed (commented):
Thanks for updating the Alert styles @psealock . I left a few comments on the PR.

The `Learn more` button font size is too big on mobile. Other than that, the changes tested well!


<img width="305" alt="Screenshot 2024-07-05 at 13 55 44" src="https://github.com/woocommerce/woocommerce/assets/4344253/c76838ec-3d94-4cc3-ac6b-3b483c21056f">

ilyasfoo reviewed (changes requested):
The new alert is looking great! The alert overflows screen in `WooCommerce > Extensions` when the browser width to `782px`.

<img width="785" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/a8bab0b6-975b-427c-8254-0e30996f0096">

I'm able to reproduce the large action buttons when browser width is `782px` or less, but there's also some inconsistencies for the `Dismiss` button. In my video, the first alert's `Dismiss` button is small, but the second and third is huge:


https://github.com/woocommerce/woocommerce/assets/3747241/0f9d8644-fd59-4492-9d8a-575735c68241

psealock committed:
Update plugins/woocommerce-admin/client/layout/store-alerts/index.js

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

psealock committed:
remove button sizes media query

psealock committed:
remove marketplace styles

psealock committed:
fix tests

psealock commented:
>The new alert is looking great! The alert overflows screen in WooCommerce > Extensions when the browser width to 782px.

Thanks @ilyasfoo, Turns out Marketplace added their own media queries to Store Alerts. Thanks for checking. I fixed this in [remove marketplace styles](https://github.com/woocommerce/woocommerce/pull/49174/commits/bc96bc14042a51a26ad881d718cb06ce57a33437).

I can't reproduce the button size issue, but I did fix the enlargening buttons for small screens. Maybe that fixed the problem?

@chihsuan, thanks for the review. I fixed the button sizes in [remove button sizes media query](https://github.com/woocommerce/woocommerce/pull/49174/commits/5b07da0c70f715158299c5814671cafeeedf0499).

chihsuan reviewed (approved):
Thanks, LGTM! 🚢

ilyasfoo commented:
> Thanks @ilyasfoo, Turns out Marketplace added their own media queries to Store Alerts. Thanks for checking. I fixed this in [remove marketplace styles](https://github.com/woocommerce/woocommerce/pull/49174/commits/bc96bc14042a51a26ad881d718cb06ce57a33437).

Thanks for addressing, @psealock! However, I'm still able to reproduce this with `< 782px` width screen in the Extensions page, where the right padding overflows:

<img width="1014" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/111831bd-7ebc-4b46-9fc0-3e896e6d6203">

Comparing to homescreen, where the right padding is visible on screen:

<img width="1009" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/9da12dee-5260-47bf-85f2-a20df583deb0">


>I can't reproduce the button size issue, but I did fix the enlargening buttons for small screens. Maybe that fixed the problem?

Nice, I can't reproduce it anymore 👍

psealock committed:
fix marketplace css, again

psealock committed:
reapply narrow margin

psealock commented:
Thanks for double checking @ilyasfoo. I think I finally got it this time

<img width="1231" alt="image" src="https://github.com/user-attachments/assets/995302b7-1253-4945-95b1-0b7c16ea102f">

ilyasfoo reviewed (approved):
Awesome, @psealock! I confirmed it is now fixed! 🚀

ilyasfoo committed:
Merge branch 'trunk' into update/store-alerts-inner-styles
